File: backend/src/seaguard/api/schemas/live.py
from datetime import datetime

from pydantic import BaseModel, Field


class LiveImportStatus(BaseModel):
    job_id: int
    source_file: str
    status: str

    rows_read: int = Field(ge=0)
    rows_imported: int = Field(ge=0)
    rows_rejected: int = Field(ge=0)
    duplicates_skipped: int = Field(ge=0)

    started_at: datetime
    finished_at: datetime | None


class LiveStatusResponse(BaseModel):
    server_time: datetime

    latest_ais_timestamp: datetime | None

    vessel_count: int = Field(ge=0)
    message_count: int = Field(ge=0)

    ingestion_active: bool

    latest_import: LiveImportStatus | None

File: frontend/src/api/live.ts
const API_BASE_URL =
  "http://127.0.0.1:8000";

export interface LiveImportStatus {
  job_id: number;
  source_file: string;
  status: string;

  rows_read: number;
  rows_imported: number;
  rows_rejected: number;
  duplicates_skipped: number;

  started_at: string;
  finished_at: string | null;
}

export interface LiveStatus {
  server_time: string;
  latest_ais_timestamp:
    | string
    | null;

  vessel_count: number;
  message_count: number;

  ingestion_active: boolean;

  latest_import:
    | LiveImportStatus
    | null;
}

export async function fetchLiveStatus(
  signal?: AbortSignal,
): Promise<LiveStatus> {
  const response = await fetch(
    `${API_BASE_URL}/api/v1/live/status`,
    {
      signal,
    },
  );

  if (!response.ok) {
    throw new Error(
      `Live status request failed: ${response.status}`,
    );
  }

  return response.json() as Promise<LiveStatus>;
}